File: CsFormsFlashcards/CsFormsFlashcards/Agregar.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using System.IO;

namespace CsFormsFlashcards
{
    public partial class Agregar : Form
    {
        public Agregar()
        {
            InitializeComponent();
        }

        private void radioButton2_CheckedChanged(object sender, EventArgs e)
        {
            if (radioButton2.Checked == true)
            {

                button2.Visible = true;
                pictureBox1.Visible = true;
            }
            else 
            {
                button2.Visible = false;
                pictureBox1.Visible = false;
            
            }
        }

        private void pictureBox1_Click(object sender, EventArgs e)
        {

        }

        private void Agregar_Load(object sender, EventArgs e)
        {
            button2.Visible = false;
        }

        string lugar = "";
        private void button2_Click(object sender, EventArgs e)
        {
            OpenFileDialog open = new OpenFileDialog();

            open.Filter = "Image Files(*.jpg)|*.jpg";
            if (open.ShowDialog() == DialogResult.OK)
            {
                pictureBox1.Image = new Bitmap(open.FileName);
                //textBox1.Text = open.FileName;
                lugar = open.FileName;              
            }
        }

        private void button1_Click(object sender, EventArgs e)
        {
            if (radioButton2.Checked == true)
            {
                string ruta = @"C:\Users\oscar\Desktop\Imagen\";
                string rutaynombre = ruta + textBox1.Text + ".jpg";
                FlashCardsArchivosImagen img = new FlashCardsArchivosImagen(lugar, rutaynombre);

                FlashCardsArchivos grd = new FlashCardsArchivos();
                grd.Guardartexto(@"C:\Users\oscar\Desktop\Puntaje\", textBox1.Text, textBox2.Text);

            }
            else
            {
                FlashCardsArchivos grd = new FlashCardsArchivos();
                grd.Guardartexto(@"C:\Users\oscar\Desktop\Puntaje\", textBox1.Text, textBox2.Text);

            }
         
 
        }
    }
}

File: VbFormsFlashcards/VbFormsFlashcards/Agregar.vb
Imports System.IO

Public Class Agregar
    Private Sub Agregar_Load(sender As Object, e As EventArgs) Handles MyBase.Load

        Button2.Visible = False

    End Sub

    Private Sub RadioButton2_CheckedChanged(sender As Object, e As EventArgs) Handles RadioButton2.CheckedChanged

        If RadioButton2.Checked = True Then

            Button2.Visible = True
            PictureBox1.Visible = True
        Else
            Button2.Visible = False
            PictureBox1.Visible = False
        End If


    End Sub


    Dim lugar As String
    Private Sub Button2_Click(sender As Object, e As EventArgs) Handles Button2.Click

        Dim open As New OpenFileDialog()
        open.Filter = "Image Files(*.jpg)|*.jpg"

        If open.ShowDialog() = DialogResult.OK Then

            PictureBox1.Image = New Bitmap(open.FileName)
            lugar = open.FileName

        End If

    End Sub

    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles Button1.Click

        If RadioButton2.Checked = True Then

            Dim ruta As String = "C:\Users\oscar\Desktop\Imagen\"
            Dim rutaynombre As String = ruta & TextBox1.Text & ".jpg"
            Dim img As New FlashCardsArchivosImagen(lugar, rutaynombre)

            Dim grd As New FlashCardsArchivos()
            grd.GuardarTexto("C:\Users\oscar\Desktop\Puntaje\", TextBox1.Text, TextBox2.Text)

        Else
            Dim grd As New FlashCardsArchivos()
            grd.GuardarTexto("C:\Users\oscar\Desktop\Puntaje\", TextBox1.Text, TextBox2.Text)

        End If


    End Sub
End Class